src/executor/sort_merge_join.rs

```diff
@@ -0,0 +1,149 @@
+            println!("{:?}", row);
+        }
+        // // collect all chunks from children
+        // let (left_chunks, right_chunks) = async {
+        //     tokio::try_join!(
+        //         self.left_child.try_collect::<Vec<DataChunk>>(),
+        //         self.right_child.try_collect::<Vec<DataChunk>>(),
+        //     )
+        // }
+        // .await?;
+
+        // // get rows iterator from chunks
+        // let left_rows = || left_chunks.iter().flat_map(|chunk| chunk.rows()).peekable();
+        // let right_rows = || {
+        //     right_chunks
+        //         .iter()
+        //         .flat_map(|chunk| chunk.rows())
+        //         .peekable()
+        // };
+
+        // // build
+        // let data_types = self.left_types.iter().chain(self.right_types.iter());
+        // let mut builder = DataChunkBuilder::new(data_types, PROCESSING_WINDOW_SIZE);
+        // let mut left_row = left_rows().next().unwrap();
+        // let mut right_row = right_rows().next().unwrap();
+        // loop {
+        //     match compare_row(
+        //         &left_row,
+        //         &right_row,
+        //         &self.left_column_indexes,
+        //         &self.right_column_indexes,
+        //     ) {
+        //         Ordering::Equal => {
+        //             let values = left_row.values().chain(right_row.values());
+        //             if let Some(chunk) = builder.push_row(values) {
+        //                 yield chunk;
+        //             }
+        //             if left_rows().peek().is_none() {
+        //                 break;
+        //             }
+        //             left_row = left_rows().next().unwrap();
+        //         }
+        //         Ordering::Greater => {
+        //             if right_rows().peek().is_none() {
+        //                 break;
+        //             }
+        //             right_row = right_rows().next().unwrap();
+        //         }
+        //         Ordering::Less => {
+        //             if left_rows().peek().is_none() {
+        //                 break;
+        //             }
+        //             left_row = left_rows().next().unwrap();
+        //         }
+        //     }
+        // }
+        // // if rows line < PROCESSING_WINDOW_SIZE ,take rest rows out of builder
+        // if let Some(chunk) = { builder }.take() {
+        //     yield chunk;
+        // }
+    }
+}
+
+pub fn compare_row(
+    left_row: &RowRef,
+    right_row: &RowRef,
+    left_column_indexes: &[usize],
+    right_column_indexes: &[usize],
+) -> Ordering {
+    let left_data_value = left_row.get_by_indexes(left_column_indexes)[0].clone();
+    let right_data_value = right_row.get_by_indexes(right_column_indexes)[0].clone();
+    left_data_value.partial_cmp(&right_data_value).unwrap()
+}
+
+#[try_stream(boxed,ok = RowRef<'static>, error = ExecutorError)]
+async fn row_stream(result: Result<DataChunk, ExecutorError>) {
```

Comment:
That's one of the most weird (and unsolved) problem of the Rust programming language -- you simply cannot return something with lifetime from a stream of fully owned data.

I'd recommend you to write in the following way:

```rust
#[try_stream(boxed, ok = Row, error = ExecutorError)]
async fn row_stream(stream: BoxedExecutor) {
    #[for_await]
    for chunk in stream {
        for row in chunk?.rows() {
            yield row.values().collect();
        }
    }
}
```

Note that this is not as efficient as before, as now we have to create a copy of each row when yielding (`row.values().collect()` creates a fully owned Row). But we can optimize it later.

We can keep everything simple in this PR, and see how to optimize things later, if we want to use `RowRef` and do zero extra allocations during the whole process.

Reply:
Thanks for such a nice reply, I just thought it was a simple lifetime problem, but I didn't expect it to be so complicated... Anyway, I'll finish this simple PR as soon as I can and come back with a better way to optimize it.